### PR TITLE
Pic 2459

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/application/Config.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.probation.courtcaseservice.application;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.time.Clock;
+
+@Configuration
+public class Config {
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/application/Config.java]: Updated, thanks

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/CourtCaseRepository.java [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 import uk.gov.justice.probation.courtcaseservice.jpa.entity.CourtCaseEntity;
 
+import java.util.List;
 import java.util.Optional;
 
 @Repository
 public interface CourtCaseRepository extends CrudRepository<CourtCaseEntity, Long>{
     Optional<CourtCaseEntity> findFirstByCaseIdOrderByIdDesc(String caseId);
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/CourtCaseRepository.java:13]: Updated, thanks

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CaseSearchResultItemMapper.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
+public class CaseSearchResultItemMapper {
+    @Autowired
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CaseSearchResultItemMapper.java]: Updated, thanks

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CaseSearchResultItemMapper.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    private final Clock clock;
+
+    public CaseSearchResultItemMapper(final Clock clock) {
+        this.clock = clock;
+    }
+
+
+    public CaseSearchResultItem from(CourtCaseEntity courtCaseEntity, final String crn) {
+
+        // filter out hearing defendants that does not match teh CRN - as the case may have multiple defendants with different CRNs
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CaseSearchResultItemMapper.java]: CaseSearchResultItemMapper

File: src/test/java/uk/gov/justice/probation/courtcaseservice/controller/CaseSearchControllerIntTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            .body("items[1].crn", equalTo("X25829"))
+            .body("items[1].probationStatus", equalTo("Current"))
+            .body("items[1].offenceTitles", equalTo(List.of("Theft from a hospital", "Theft from a shop")))
+            .body("items[1].lastHearingDate", equalTo(LocalDate.now().minusDays(2).format(DateTimeFormatter.ISO_DATE)))
+            .body("items[1].lastHearingCourt", equalTo("Sheffield"))
+            .body("items[1].nextHearingDate", equalTo(LocalDate.now().plusDays(10).format(DateTimeFormatter.ISO_DATE)))
+            .body("items[1].nextHearingCourt", equalTo("Leicester"))
+            .body("items[1].awaitingPsr", equalTo(true))
+            .body("items[1].breach", equalTo(true))
+            ;
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/controller/CaseSearchControllerIntTest.java]: This has been a practice in all such JSON body assertions, probably to easily duplicate the last line and update. 
Addressed it anyway

File: src/test/java/uk/gov/justice/probation/courtcaseservice/service/CaseSearchServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        verify(courtCaseRepository).findAllCasesByCrn(TEST_CRN);
+        verify(caseSearchResultItemMapper, times(2)).from(caseArgCapture.capture(), crnArgCapture.capture());
+        assertThat(caseArgCapture.getAllValues()).isEqualTo(List.of(case1, case2));
+        assertThat(crnArgCapture.getAllValues()).isEqualTo(List.of(TEST_CRN, TEST_CRN));
+
+        assertThat(actual).isEqualTo(CaseSearchResult.builder().items(List.of(result1, result2)).build());
+    }
+
+    @Test
+    void shouldReturn() {
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/service/CaseSearchServiceTest.java]: Updated, thanks

**Review by vb-moj — APPROVED** [review]

**Review by vb-moj — APPROVED** [review]

**Pull request merged**